### PR TITLE
enh: custom UI for scoped context chat

### PR DESCRIPTION
**Pull request by kyteinsky**
Design feedback required on the selector UI (Select File/Folder-Select Providers). It looks a bit too big imo. The same selector is present in speech-to-text UI so that would also need to be changed to maintain consistency. Something less obstructing/with less occupancy on the dialog is what I imagine would be a good fit.

@nimishavijay @jancborchardt 

image

### CODE REVIEW

**Review by jancborchardt — COMMENTED** [review]
Scoped Context Chat and Context Chat don't need to be separate entries, they could be the same?

And Context Chat could just have a button on the right of the heading+subline saying e.g. "Pick folder or storage" which then opens an action menu.

@nimishavijay for more specific proposals with mockup, if you have firther ideas? :)

File: src/components/AssistantFormInputs.vue [hunk 1]
```diff
@@ -256,17 +289,31 @@ export default {
 				},
 			)
 		},
+		onUpdateCc() {
```
Comment (julien-nc) [inline src/components/AssistantFormInputs.vue:292]:
```suggestion
		onUpdateContextChat() {
```
Maybe more explicit names are needed here.

File: src/components/AssistantFormInputs.vue [hunk 2]
```diff
@@ -256,17 +289,31 @@ export default {
...
+				},
+			)
+		},
 	},
 }
 
 </script>
 
-<style land="scss" scoped>
+<style lang="scss" scoped>
```
Comment (julien-nc) [inline src/components/AssistantFormInputs.vue:309]: :+1:  :see_no_evil:

File: lib/Service/AssistantService.php [hunk 3]
```diff
@@ -132,6 +133,28 @@ private function sanitizeInputs(string $type, array $inputs): array {
...
+
+					if (count($inputs) === 3) {
+						if (!isset($inputs['scopeType']) || !is_string($inputs['scopeType'])
+							|| !isset($inputs['scopeList']) || !is_array($inputs['scopeList'])) {
+							throw new \Exception('Invalid input(s)');
+						}
+						$inputs['scopeList'] = array_filter(
+							array_map(fn (array $scope) => $scope['id'] ?? null, $inputs['scopeList']),
+							fn ($scope) => is_string($scope),
+						);
```
Comment (julien-nc) [inline lib/Service/AssistantService.php:155]: Will the scope list be loaded correctly when displaying an existing task?
Wouldn't it be simpler to keep the structure of `$inputs['scopeList']` intact so it is straightforward to load it in `<ContextChatInputForm>`?

File: src/components/AssistantFormInputs.vue [hunk 4]
```diff
@@ -57,6 +57,12 @@
 		</div>
 	</div>
 	<div v-else class="assistant-inputs">
+		<div v-if="selectedTaskTypeId === 'OCA\\ContextChat\\TextProcessing\\ContextChatTaskType'" class="assistant-inputs">
+			<NcCheckboxRadioSwitch :checked.sync="sccEnabled">
+				{{ t('assistant', 'Reduce context') }}
+			</NcCheckboxRadioSwitch>
+			<ContextChatInputForm v-if="sccEnabled" v-model="sccData" />
```
Comment (julien-nc) [inline src/components/AssistantFormInputs.vue:64]:
```suggestion
			<ContextChatInputForm v-if="sccEnabled" :scc-data.sync="sccData" />
```
But maybe you had good reasons to use v-model.

**Review by julien-nc — COMMENTED** [review]
LGTM, a few questions.

File: lib/Service/AssistantService.php (hunk 3 above)
Comment (kyteinsky) [reply lib/Service/AssistantService.php:155]: hmm, it did work in the Assistant modal but noticed just now that it does not when pulled from the result notification. Will de-couple the UI and actual data shortly.

File: src/components/AssistantFormInputs.vue (hunk 4 above)
Comment (kyteinsky) [reply src/components/AssistantFormInputs.vue:64]: not really, would switch to this for consistency. Just discovered that they do the same thing.

**Comment by AndyScherzinger** [comment]
Just as a side note, can't tell where mimetype icons are fetched from, we are currently in the process of updating the icons throughout the system(s) server and clients, see https://github.com/nextcloud/server/issues/44340 targeting the v29.0.0 release - fyi @kyteinsky @julien-nc

**Comment by kyteinsky** [comment]
@jancborchardt new screenshots with the suggested changes:

Screenshot from 2024-03-22 15-14-53
Screenshot from 2024-03-22 15-14-40

@AndyScherzinger Thanks for the heads up! I think the only svg icon we're using is folder.svg from `/apps/theming/img/core/filetypes/folder.svg`. This seems unaffected/the latest. Others are procured from the php side using the preview manager. So we're good.

**Review by julien-nc — APPROVED** [review]

**Pull request merged**